### PR TITLE
[plugin.video.embycon] 1.9.18

### PR DESCRIPTION
**Pull request by faush01**
### Description
July Release
A lot of small fixes and compatibility changes for the latest version of the Emby 4.2 Server.

I hope this is the last Python 2.7 release!
I am waiting for nightly Kodi 19 builds with Python 3 to switch my attention to that, hopefully it is soon.

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

**Comment by faush01** [comment]
The above are not errors for Python 2.7

**Comment by basrieter** [comment]
It has to be compatible with both Py2 and Py3: 

> After the release of Kodi 18 (Leia) only addons (new and/or updates) that are compatible with both Python 2 and 3 or Python 3-only will be accepted to the repositories

See: https://forum.kodi.tv/showthread.php?tid=327746

**Comment by faush01** [comment]
I am happy to start work on a switch to Python 3 but I am not going to be working on Python 2 and 3 compatibility in the same code. Too many headaches.

take the above example for instance:
iteritems() and items() do different things on Python2, iteritems() does what items() does on Python3, I am not going to be spending time on testing how this effect performance on multiple platforms especially when this code base will never touch Python3.

As I said I am happy to start work on Python 3 when there are nightly builds from main that have Python3 in them but until then I will be sticking with Kodi 18 Python 2.

**Comment by faush01** [comment]
I will close this and submit a deprecation version

File: plugin.video.embycon/resources/lib/websocket.py [hunk 1]
```diff
@@ -894,8 +906,13 @@ def run_forever(self, sockopt=None, sslopt=None, ping_interval=0):
                     self._callback(self.on_message, data)                    
                     
                 except Exception as e:
-                    #print str(e.args[0])
-                    if "timed out" not in e.args[0]:
+                    found_timeout = False
+                    for arg in e.args:
+                        if isinstance(arg, str):
+                            print str(arg)
```
Comment (Rechi) [inline plugin.video.embycon/resources/lib/websocket.py]: The following change would have been enough to get CI green.
```suggestion
                            print(str(arg))
```

Comment (faush01) [reply plugin.video.embycon/resources/lib/websocket.py]: But it is always something, I come to dread having to try to get the addon updated, it is not worth the stress and issues.

Comment (faush01) [reply plugin.video.embycon/resources/lib/websocket.py]: Nothing against the project, it is just not for me, I am going to go back to a private repo, it is just so much easier to manage and update.

**Comment by faush01** [comment]
@Rechi 
I think I just need to understand what is going on.
Why is there a requirement for both Pthon 2 and 3 compatibility, yes you are switching from 2 to 3 some time in the future but the current Kodi 18 is Python 2 and that is what the addon is targeting. I dont see all the fuss about concurrent Python 2 and 3 compatibility, when Kodi 19 has Python 3 then people can start testing and working out how to best support both version or just make a clean cut to Python 3.
This should be left up to the addon devs, it should not be forced on them.
We don't even have Kodi 19 nightly build with Python 3 to test on yet.

Its not just the print() lines, the iterator changes were also failing in the build, the problem is the changes requested would potentially fix a Python 2 and 3 dual comparability requirements but the items() and iteritems() does different things in Python 2.
Also it looks like the verification you are using is only checking the changed lines, so this will not catch all compatibility issues.

As I have said, once Kodi 19 has nightly builds that have Python 3 then I will start looking at it but until then there is no point. As a plugin dev I feel that should be my call.

Please help me understand.

**Comment by Rechi** [comment]
As your add-on is in the krypton branch you are currently not forced to be completely Python 3 compatible (although it would be nice if you are already).

There are a a few checks (old Python 2 except, exec, ne, print, raise, repr and tuple_params syntax) which fail the build as the new Python 3 syntax also works in Python 2. The others, like items/iteritems, are only reported as information and don't fail the check as they are not always needed.

**Comment by faush01** [comment]
@Rechi 
I remember fixing all the old Python 2 checks a few versions ago but though I was fine now until Kodi 19.

But the items/iteritems was breaking the build, I will create a new PR and try again.

**Comment by Rechi** [comment]
Trust me, items/iteritems didn't break the build. Only the print line caused the failure, which was previously commented out.

**Comment by faush01** [comment]
Ok thanks for the help, I will try again.

**Pull request closed without merging**